syntax/eval.go

```diff
@@ -1265,13 +1265,15 @@ func (e *Expr) evalComprK(sess *Session, env *values.Env, ident string, begin in
 		// TODO: make sure we compute the same digest for the single-clause case.
 		//	(and test this)
```

Comment:
Hi @mariusae, if you're available I'd really appreciate your input on this bug fix. 

Also, I don't quite understand this TODO, any chance you know what it means? Is it saying that the digest for all single-clause comprehensions should be the same (that doesn't seem right)?